dropshot_endpoint/src/api_trait.rs

```diff
@@ -154,13 +157,61 @@ impl ApiMetadata {
     }
 }
 
+/// A mirror of dropshot's `TagConfig`, used as part of arguments to the
+/// top-level `api_description` macro.
+#[derive(Deserialize, Debug)]
+struct ApiTagConfig {
+    #[serde(default)]
+    allow_other_tags: bool,
+    #[serde(default)]
+    endpoint_tag_policy: ApiEndpointTagPolicy,
+    // tag_definitions is required
+    tag_definitions: HashMap<String, ApiTagDetails>,
```

Comment:
how about `tags` ? I think it would be ok to change `dropshot::TagConfig` if we want them

Reply:
> I think it would be ok to change dropshot::TagConfig if we want them

Yeah I was thinking about suggesting that, I think I value consistency in names highly enough that if we're changing to `tags` here (something I think is a good idea) we should also change `dropshot::TagConfig`.

All right, I'll make that change.

Reply:
Done.